File: src/java/org/elastos/hive/DataStorage.java
package org.elastos.hive;

public interface DataStorage {
	String loadBackupCredential(String serviceDid);
	String loadAccessToken(String serviceDid);
	String loadAccessTokenByAddress(String providerAddress);

	void storeBackupCredential(String serviceDid, String credential);
	void storeAccessToken(String serviceDid, String accessToken);
	void storeAccessTokenByAddress(String providerAddress, String accessToken);

	void clearBackupCredential(String serviceDid);
	void clearAccessToken(String serviceDid);
	void clearAccessTokenByAddress(String providerAddress);
}

File: src/utils/storage/datastorage.ts
/**
 * The data storage for the access token, the backup credential, etc. which comes from the hive node.
 */
export interface DataStorage {
	loadBackupCredential(serviceDid: string): Promise<string>;
	loadAccessToken(serviceDid: string): Promise<string>;
	loadAccessTokenByAddress(providerAddress: string): Promise<string>;

	storeBackupCredential(serviceDid: string, credential: string): Promise<void>;
	storeAccessToken(serviceDid: string, accessToken: string): Promise<void>;
	storeAccessTokenByAddress(providerAddress: string, accessToken: string): Promise<void>;

	clearBackupCredential(serviceDid: string): Promise<void>;
	clearAccessToken(serviceDid: string): Promise<void>;
	clearAccessTokenByAddress(providerAddress: string): Promise<void>;
}
